cmd/kube-proxy/app/server.go

```diff
@@ -237,6 +237,9 @@ func NewProxyServerDefault(config *options.ProxyServerConfig) (*ProxyServer, err
 		glog.V(0).Info("Tearing down pure-iptables proxy rules.")
 		iptables.CleanupLeftovers(iptInterface)
 	}
+	if err != nil && proxier == nil {
```

Comment:
My point is mostly that if err was ever nil, we should have hit a fatal error already.  If we got here with a nil proxier, someone returned no error and a nil value.  I would be fine with a check for nil value here and a fmt.Errorf() explaining it, but this && can never ever trigger, unless I am missing somethign...